allennlp/service/servable/servable.py

```diff
@@ -0,0 +1,59 @@
+from typing import Dict, Any, Optional, List
+
+JSONDict = Dict[str, Any]  # pylint: disable=invalid-name
+
+
+class Servable:
+    """
+    Any "model" that we want to serve through either our command line tool or REST API / webapp
+    needs to implement this interface.
+    """
+    def predict_json(self, inputs: JSONDict) -> JSONDict:
+        raise NotImplementedError()
+
+    # TODO(joelgrus): add a predict_tensor method maybe?
+
+    @classmethod
+    def from_config(cls, config: JSONDict = None) -> 'Servable':  # pylint: disable=unused-argument
+        # default implementation calls no-argument constructor, you probably want to override this
+        return cls()
+
+
+class ServableCollection:
```

Comment:
The `Registry` is also the way that it is so that third-party users can add things to it and still use our experiment framework.  I don't think we're trying to solve that problem with the service, which is another argument against merging them.  As I said, I could go either way here.